lib/corefoundation/base.rb

```diff
@@ -33,11 +33,12 @@ def self.typecast(cftyperef)
     def initialize(pointer)
       @ptr = FFI::Pointer.new(pointer)
       ObjectSpace.define_finalizer(self, self.class.finalize(ptr))
+      CF.retain(ptr)
```

Comment:
This change would result in `.retain` being called twice for certain cases, could you check the behaviour for that and prevent it (if required)?

Reply:
I am doubtful, putting an all encompassing retain in constructor and removing external `.retain` calls, is the way to go. I would recommend tracking this in an issue.